mysql-test/suite/innodb/t/percona_session_temp_tablespaces_encrypt.test

```diff
@@ -0,0 +1,109 @@
+INSERT INTO t2 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+
+INSERT INTO t3 SELECT * FROM t1;
+INSERT INTO t3 SELECT * FROM t2;
+INSERT INTO t3 SELECT * FROM t3;
+INSERT INTO t3 SELECT * FROM t3;
+
+--replace_column 1 ID
+--replace_regex /\\#innodb_temp\\temp/\/#innodb_temp\/temp/
+SELECT * FROM INFORMATION_SCHEMA.INNODB_SESSION_TEMP_TABLESPACES WHERE ID = connection_id() ORDER BY SPACE;
+
+SET GLOBAL innodb_encrypt_tables=ON;
+
+CREATE TEMPORARY TABLE test.t4(a INT, b BLOB);
+CREATE TEMPORARY TABLE test.t5(a INT, b BLOB);
+CREATE TABLE t6 like t1;
+INSERT INTO t4 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+INSERT INTO t5 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+INSERT INTO t6 SELECT * FROM t4;
+INSERT INTO t6 SELECT * FROM t5;
+INSERT INTO t6 SELECT * FROM t6;
+INSERT INTO t6 SELECT * FROM t6;
+
+--replace_column 1 ID
+--replace_regex /\\#innodb_temp\\temp/\/#innodb_temp\/temp/
+SELECT * FROM INFORMATION_SCHEMA.INNODB_SESSION_TEMP_TABLESPACES WHERE ID = connection_id() ORDER BY SPACE;
+SET GLOBAL innodb_encrypt_tables=OFF;
+DROP TABLE t3, t6;
+
+SET GLOBAL innodb_temp_tablespace_encrypt=ON;
+SELECT @@innodb_encrypt_tables;
+--echo # Create a new connection.
+connect (con2, localhost, root,,);
+connection con2;
+SELECT @@session.big_tables;
+
+--echo # Ensure entries in information_schema.processlist and
+--echo # information_schema.innodb_session_temp_tablespaces are
+--echo # in sync
+
+CREATE TEMPORARY TABLE test.t1(a INT, b BLOB);
+CREATE TEMPORARY TABLE test.t2(a INT, b BLOB);
+CREATE TABLE t3 like t1;
+INSERT INTO t1 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+INSERT INTO t2 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+
+INSERT INTO t3 SELECT * FROM t1;
+INSERT INTO t3 SELECT * FROM t2;
+INSERT INTO t3 SELECT * FROM t3;
+INSERT INTO t3 SELECT * FROM t3;
+
+--replace_column 1 ID
+--replace_regex /\\#innodb_temp\\temp/\/#innodb_temp\/temp/
+SELECT * FROM INFORMATION_SCHEMA.INNODB_SESSION_TEMP_TABLESPACES WHERE ID = connection_id() ORDER BY SPACE;
+
+SET GLOBAL innodb_encrypt_tables=ON;
+
+CREATE TEMPORARY TABLE test.t4(a INT, b BLOB);
+CREATE TEMPORARY TABLE test.t5(a INT, b BLOB);
+CREATE TABLE t6 like t1;
+INSERT INTO t4 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+INSERT INTO t5 values (1, 'hello'), (2, 'hi'), (3, 'wl11613'), (4, 'temp'), (5, 'tablespace');
+INSERT INTO t6 SELECT * FROM t4;
+INSERT INTO t6 SELECT * FROM t5;
+INSERT INTO t6 SELECT * FROM t6;
+INSERT INTO t6 SELECT * FROM t6;
+
+--replace_column 1 ID
+--replace_regex /\\#innodb_temp\\temp/\/#innodb_temp\/temp/
+SELECT * FROM INFORMATION_SCHEMA.INNODB_SESSION_TEMP_TABLESPACES WHERE ID = connection_id() ORDER BY SPACE;
+
+connection default;
+disconnect con1;
+disconnect con2;
+
+SET GLOBAL innodb_temp_tablespace_encrypt=default;
```

Comment:
Later: do not reset to default, but save to user var and restore

Reply:
afaik, every test has to match the values at beginning of check testcases (which has default values). So I think default is fine.

check-testcase also does this.

Reply:
Not if MTR is started with --mysqld=--var=non-default